Lib/zipfile.py

```diff
@@ -566,6 +566,49 @@ def decrypter(data):
     return decrypter
 
 
+# Supports a password-based encryption.
+#
+# Usage:
+#     zd = _ZipEncrypter(mypwd)
+#     cypher_bytes = zd(plain_bytes)
+
+def _ZipEncrypter(pwd):
+    key0 = 305419896
+    key1 = 591751049
+    key2 = 878082192
+
+    global _crctable
+    if _crctable is None:
+        _crctable = list(map(_gen_crc, range(256)))
+    crctable = _crctable
+
+    def crc32(ch, crc):
+        """Compute the CRC32 primitive on one byte."""
+        return (crc >> 8) ^ crctable[(crc ^ ch) & 0xFF]
+
+    def update_keys(c):
+        nonlocal key0, key1, key2
+        key0 = crc32(c, key0)
+        key1 = (key1 + (key0 & 0xFF)) & 0xFFFFFFFF
+        key1 = (key1 * 134775813 + 1) & 0xFFFFFFFF
+        key2 = crc32(key1 >> 24, key2)
+
+    for p in pwd:
+        update_keys(p)
+
+    def encrypter(data):
+        """Encrypt a bytes object."""
+        result = bytearray()
+        append = result.append
+        for c in data:
+            k = key2 | 2
+            update_keys(c)
+            c ^= (((k * (k^1)) >> 8) & 0xFF)
+            append(c)
+        return bytes(result)
```

Comment:
Since this is only used internally, is the conversion to `bytes()` necessary? Doing this conversion likely means the entire data is copied.